src/locale/localed.c

```diff
@@ -395,6 +399,37 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
         for (LocaleVariable p = 0; p < _VARIABLE_LC_MAX; p++)
                 free_and_replace(c->locale[p], new_locale[p]);
```

Comment:
hmm, i think this should be moved after the generating loop you add below. I.e. this makes the data passed in the data localed now thinks is correct, but if the loop you added fails it shouldn't be made the new "correct" data. i.e. your loop should operate on new_locale[] still, and only once your loop passed we should move new_locale[p] into c->locale[] to take it effect. After all the localegen stuff you add is preparation to make the locale setting count, if it fails we should not have the new settings take effect.